### ISSUE TITLE
Wrong position of `string-quotes` offenses with code comments

### ISSUE THREAD

**Issue by AlexWayfer** [stylelint/stylelint#6314]
### What steps are needed to reproduce the bug?

```scss
@mixin small-and-large-elements(
	/*
	Probably here should be something like `$large_element_container`,
	for example, for Cccombo input like this:
	https://gitlab.gettransport.com/site/site/merge_requests/80/diffs#28551ac9184a7e08499c6a0b6a17f78126a4defa_124_124
	Discussion:
	https://gitlab.gettransport.com/site/site/merge_requests/522#note_324681
	*/
	$small_element: '> button', $large_element: '> input',
	$small_element_on_right: false, $inline: false
) {
}
```

(with `//` the same offenses)

### What Stylelint configuration is needed to reproduce the bug?

```yaml
extends:
  - stylelint-config-standard
  - stylelint-config-standard-scss

ignorePath: .gitignore

plugins:
  - stylelint-no-unsupported-browser-features
```

### How did you run Stylelint?

CLI or Atom Editor plugin. 

### Which version of Stylelint are you using?

`14.11.0`

### What did you expect to happen?

No offenses about quotes inside comment.

### What actually happened?

```
assets/styles/lib/_small-and-large-elements.scss
 3:16  ✖  Expected double quotes  string-quotes
 3:44  ✖  Expected double quotes  string-quotes
```

![image](https://user-images.githubusercontent.com/6510020/188022448-f0be87e0-c8f8-4924-a174-98c082364f7c.png)

### Does the bug relate to non-standard syntax?

SCSS, but not sure.

**Comment by AlexWayfer** [comment]
Oh, I got it: it just calculates the wrong position (via CLI too):

![image](https://user-images.githubusercontent.com/6510020/188022748-928ee710-5cc5-49db-bfe9-44651929c85d.png)

**Comment by ybiquitous** [comment]
@AlexWayfer Thanks for using the template and providing the details. We can reproduce the behavior on the [demo](https://stylelint.io/demo#N4Igxg9gJgpiBcIACBbAlgDzQOwAQGcUBDAGxIFojspySiAnAcxnJhJhRmwBd8AKADrYB3APQAqISIAK9CACMi8kgE9cACxj0YBdRACuJKLnk78ETt3U5GuEmgDWOgAYASOkxgB9Nhy7cvSB4iHC1nABopbgAzCHpcGAwiFAAHdnDcWPiAYTBIFHkIXBwU-W47Rx0rNHx4KPVubhTa0VFGNG46eQA6Zkb6KnwUuO5u-NF8DpgJqdFOTy9tAEd9GHxeUQAOAAZRKDRo6PwAYgAmTYBWC4BGIjAATmvNgBYiAHYYbZf7+7AANiI23kAOub2ib0211OAOesGiRC8UOeiNOzyiABEamB9PhJhBsHVhFZGs14K12p0lL0YP1BsN6KNxpNuNNmdN5sxFjAVmsNhdTqdjtgICyvABmVF-SFRcSiKKuQikEg+dicHjwXAAcgAfCYytx8ZqMu4GJzfGruBqdcVsKVuEb5YqyCq-DwvPjFmhGA0NfCSPgYMacPZsDBfaQA0IAJS4YBCAC+IHCIHwKmCGAQKbAuKT4Hx0S9mbjeFwAhA9EMazLGuLuDrpZT3HoNnIKxFVYQDagBmUMDLQjr8YTIHjQA).

However, I'm afraid we have decided [not to support stylistic rules anymore](https://stylelint.io/user-guide/rules/list#stylistic-issues). Please consider switching to a pretty printer like [Prettier](https://prettier.io/) or disable the rule like this:

```scss
/* stylelint-disable string-quotes */
@mixin small-and-large-elements(
  /* ... */
);
```

**Comment by AlexWayfer** [comment]
> Please consider switching to a pretty printer like [Prettier](https://prettier.io/)

I don't want a formatter, I like linters, so it's not an alternative. OK, I'm gonna to disable the rule, thanks.
